packages/tpetra/core/src/Tpetra_CrsMatrix_decl.hpp

```diff
@@ -4788,6 +4786,18 @@ namespace Tpetra {
     Kokkos::View<impl_scalar_type*, execution_space, Kokkos::MemoryUnmanaged>
     getRowViewNonConst (const RowInfo& rowInfo) const;
 
+
+  protected:
+
+    // Friend the tester for CrsMatrix::swap
+    template <class LO, class GO, class N> friend class Tpetra::crsMatrix_Swap_Tester;
```

Comment:
Questions to the experts about syntax:

Should this be 
`friend class Tpetra::crsMatrix_Swap_Tester<LocalOrdinal, GlobalOrdinal, Node>;`
(without the template specification at the beginning of the line)?

And why are ordinal and node templates needed, but scalar template is not?  Will this test break if I build Trilinos with only float scalars (not double scalars)?

Reply:
Fixed the friend and set up the CrsMatrix test to use the SLGN test for the 4 template parameters.